server/src/main/java/org/elasticsearch/common/io/stream/StreamOutput.java

```diff
@@ -751,6 +764,16 @@ public final void writeOptionalInstant(@Nullable Instant instant) throws IOExcep
         })
     );
 
+    public <T> void writeListGeneric(List<T> v, Writer<T> writer) throws IOException {
```

Comment:
I prefer `writeGenericXXX`. You will see in current master there is `writeGenericMap`, and it also parallels with `writeGenericValue`.

Reply:
++ renamed